### ISSUE TITLE
Add an `ariaLabel` property to TreeItem

### ISSUE THREAD

**Issue by eamodio** [microsoft/vscode#95360]
Refs: #92342, https://github.com/microsoft/vscode/issues/92342#issuecomment-610076915, #95359

While implementing #92342 for the Timeline view, and thinking about #95359, it seems that the same is true for custom tree views as well. Where a `TreeDataProvider` could provide a richer value than the current (which uses the tooltip if it exists and label if not). We of course could fallback to reasonable defaults when this new property doesn't exist.

/cc @jrieken @isidorn @alexr00

**Comment by isidorn** [comment]
Agree. Same what I said in the other issue [here](https://github.com/microsoft/vscode/issues/95359#issuecomment-614170811). This makes perfect sense and we should do it.
We should also consider to potentially do the same for status bar items. fyi @bpasero

**Comment by isidorn** [comment]
I think this realisitcly will not be done in April, thus pushing to May so we do it start of milestone.
Also we would to add aria-label for StatuBarItems thus also assigning to me

**Comment by jrieken** [comment]
Let's also keep in mind that ariaLabel is likely insufficient and that we should eye at a complex object that allows to define label, description and potentially more. Unsure if aria is the right term tho?

**Comment by isidorn** [comment]
The whole W3spec calls it aria https://w3c.github.io/aria/
It stands for "accessbiilty rich internet applications".
So aria seems like a good fit. Alternative is `accessibility`.

Apart from label, we should have role and keep an open ended object to add potential stuff in the future.

**Comment by isidorn** [comment]
@jrieken I plan to join the next API meeting to discuss this.

**Comment by isidorn** [comment]
Feedback from meeting:

1. The field should be called either `Accessibility`, `AccessibilityInformation` or in the likes of that 
2. This should also be done on the `TreeItem`, `StatusBarItem` and `TimelineItem`
3. Just add to `vscode.proposed.d.ts`

After that is added to the `vscode.d.ts` I plan to pick it up in the `TimelineView`, `CustomTree` and `StatusBarItems`.
Another feedback was that the default aria label computation when `Accessibility` is not provided could be better.

Feedback on naming in 1. is very welcome. My personal preference is just `Accessibility` due to simplicity. Though `AccessibilityInformation` also works.

**Comment by isidorn** [comment]
We have added this to proposed API. Thank you very much for feedback.
Thus moving this to next milestone to do the API finalisation if all works well then.

**Comment by isidorn** [comment]
We would like to finalize this API in the june milestone however before that it would be cool if some extension starts using it. I will probably join the API meeting tomorrow so we discuss this @jrieken 

Also would be great to get some feedback for the TimelineItem from @eamodio 
I know TreeItem @alexr00 likes the changes, and for me it works for the `StatusBarItem`

**Comment by eamodio** [comment]
Looks good to me -- my only suggestion would be to call the property `accessibility` rather than `accessibilityInformation`, but I don't feel too strongly either.

**Comment by isidorn** [comment]
I have moved the `AccessibilityInformation` into `vscode.d.ts` so now it is finalised.
Test plan item created https://github.com/microsoft/vscode/issues/100936 @eamodio please note that I did not point to a TimelineItem sample in the test plan item because I could not find it. Feel free to edit the test plan item if you would like that to be covered better.
I have also improved the default label value when the ariaLabel is not provided. We extract the octicon name properly now.
